MapboxGLAndroidSDK/src/cpp/CMakeLists.txt

```diff
@@ -1,208 +1,212 @@
+            ${PROJECT_SOURCE_DIR}/style/layers/raster_layer.hpp
+            ${PROJECT_SOURCE_DIR}/style/layers/symbol_layer.cpp
+            ${PROJECT_SOURCE_DIR}/style/layers/symbol_layer.hpp
+            ${PROJECT_SOURCE_DIR}/style/layers/location_indicator_layer.cpp
+            ${PROJECT_SOURCE_DIR}/style/layers/location_indicator_layer.hpp
+            ${PROJECT_SOURCE_DIR}/style/light.cpp
+            ${PROJECT_SOURCE_DIR}/style/light.hpp
+            ${PROJECT_SOURCE_DIR}/style/position.cpp
+            ${PROJECT_SOURCE_DIR}/style/position.hpp
+            ${PROJECT_SOURCE_DIR}/style/sources/custom_geometry_source.cpp
+            ${PROJECT_SOURCE_DIR}/style/sources/custom_geometry_source.hpp
+            ${PROJECT_SOURCE_DIR}/style/sources/geojson_source.cpp
+            ${PROJECT_SOURCE_DIR}/style/sources/geojson_source.hpp
+            ${PROJECT_SOURCE_DIR}/style/sources/image_source.cpp
+            ${PROJECT_SOURCE_DIR}/style/sources/image_source.hpp
+            ${PROJECT_SOURCE_DIR}/style/sources/raster_dem_source.cpp
+            ${PROJECT_SOURCE_DIR}/style/sources/raster_dem_source.hpp
+            ${PROJECT_SOURCE_DIR}/style/sources/raster_source.cpp
+            ${PROJECT_SOURCE_DIR}/style/sources/raster_source.hpp
+            ${PROJECT_SOURCE_DIR}/style/sources/source.cpp
+            ${PROJECT_SOURCE_DIR}/style/sources/source.hpp
+            ${PROJECT_SOURCE_DIR}/style/sources/unknown_source.cpp
+            ${PROJECT_SOURCE_DIR}/style/sources/unknown_source.hpp
+            ${PROJECT_SOURCE_DIR}/style/sources/vector_source.cpp
+            ${PROJECT_SOURCE_DIR}/style/sources/vector_source.hpp
+            ${PROJECT_SOURCE_DIR}/style/transition_options.cpp
+            ${PROJECT_SOURCE_DIR}/style/transition_options.hpp
+            ${PROJECT_SOURCE_DIR}/style/value.cpp
+            ${PROJECT_SOURCE_DIR}/style/value.hpp)
 
 target_include_directories(
   mapbox-gl
-  PRIVATE
-  ${PROJECT_SOURCE_DIR}
-    ../../../vendor/mapbox-gl-native/src
-    ../../../vendor/mapbox-gl-native/platform/android/src
-    ../../../vendor/mapbox-gl-native/platform/default/include
+  PRIVATE ${PROJECT_SOURCE_DIR}
+          ../../../vendor/mapbox-gl-native/src
+          ../../../vendor/mapbox-gl-native/platform/android/src
+          ../../../vendor/mapbox-gl-native/platform/default/include)
 
-)
+target_compile_options(mapbox-gl
+                       PRIVATE $<$<CONFIG:Release>:-Oz>
+                               $<$<CONFIG:Release>:-Qunused-arguments>
+                               $<$<CONFIG:Release>:-flto>
+                               $<$<CONFIG:Release>:-fvisibility=hidden>
+                               $<$<CONFIG:Release>:-fvisibility-inlines-hidden>)
 
-target_link_libraries(
-  mapbox-gl
-  PRIVATE
-        mbgl-compiler-options
-        Mapbox::Base::jni.hpp
-        mbgl-core
-        mbgl-vendor-unique_resource
-
-)
+target_link_libraries(mapbox-gl
+                      PRIVATE $<$<CONFIG:Release>:-O2>
```

Comment:
-Os ?

Reply:
Right.

Reply:
Was it fixed in NDK? There was a long standing bug that -Os and -Oz did not work, 🤞 if -Oz works.

Reply:
Well, for ` mbgl-core`  -Oz for object files and -O2 for linker works. Let's see -Oz combination for both. :-)